app/controllers/users_controller.rb

```diff
@@ -16,7 +23,32 @@ def create
   end
   
   def show
-    @user = User.find_by_token params[:id]
+    @user = User.find_by(token: params[:id])
+  end
+
+  def update
+    @user = User.find_by(token: params[:id])
+    if @user.update(user_params)
+      flash[:notice] = "Your account is updated."
+    else
+      flash[:error] = "Something went wrong. Unsuccessful update."
+    end
+    redirect_to login_path
+  end
+
+  private
+  def user_params
+    params.require(:user).permit(:email_address, :password, :full_name)
+  end
+
+  def handle_invitation
+    if params[:token]
+      invitation = Invitation.find_by(token: params[:token])
+      inviter = invitation.user
+      Relationship.create(user: @user, follower: inviter)
+      Relationship.create(user: inviter, follower: @user)
+      invitation.update_column(:token, nil)
```

Comment:
I would push this to the model layer - `.update_column(:token, nil)` - create some instance method like `#remove_token!`